plugins/connection/aws_ssm.py

```diff
@@ -301,10 +306,10 @@ def start_session(self):
             raise AnsibleError("failed to find the executable specified %s."
                                " Please verify if the executable exists and re-try." % executable)
 
-        profile_name = ''
+        profile_name = self.get_option('profile')
```

Comment:
```suggestion
        profile_name = self.get_option('profile') or ''
```
To restore the original behavior if `profile` is not specified. On the other hand, this variable was never used, so keeping it as-is is probably better.

Reply:
As there's no prior use, I'll opt to keep it as-is.

Reply:
While the option wasn't passed it's consumed by the `cmd = [` a couple of lines below.  Switching and empty string out for None triggers a nice big exception

What I did to work around this was 
```
profile_name = '' if profile_name is None else profile_name
```
on the line above `cmd =`

Reply:
I think my solution above would be even nicer (i.e. more Pythonic) :)

Reply:
Added the more pythonic solution, thank you for catching this!

I think I didn't run into this because I was foolishly testing this across two different laptops, and the first version still had a default value set at the top - and I forgot to pull...